skimage/filters/rank/tests/test_rank.py

```diff
@@ -1,15 +1,34 @@
 import os
 import numpy as np
-from skimage._shared.testing import assert_equal
+from skimage._shared.testing import (assert_equal, assert_array_equal,
+                                     assert_allclose)
 from skimage._shared import testing
 
 import skimage
 from skimage import img_as_ubyte, img_as_float
 from skimage import data, util, morphology
 from skimage.morphology import grey, disk
 from skimage.filters import rank
+from skimage.filters.rank import __all__ as all_rank_filters
 from skimage._shared._warnings import expected_warnings
-from skimage._shared.testing import test_parallel, xfail, arch32
+from skimage._shared.testing import test_parallel, arch32, parametrize, xfail
+from pytest import param
+
+
+def test_otsu_edge_case():
+    # This is an edge case that causes OTSU to appear to misbehave
```

Comment:
Stefan please see https://github.com/scikit-image/scikit-image/issues/2528#issuecomment-432278657

Reply:
I saw that, but I don't understand: "compiler-dependent" yes, but "rounding error"—if a compiler rounds in error, then we should not return the incorrect result.

Reply:
Yeah sorry, I don't know what the cause really is, I just know both answers are mathematically close to 1E-15. So depending on the specific  circumstances, one of the two values will be chosen. I rephrased my original comment:
```
An error on the order of 1E-15 (floating64 point precision) is causing the algorithm threshold to change between two equivalent values, 41 and 81. This can be caused maybe by:
1. A decision to reorder multiplying or dividing by a constant.
2. A 32bit on 64 bit host emulation error????
3. Some optimization the compiler is doing???
4. A hardware error? doubtful? but maybe
```